### PR TITLE
Search SDK: Various bug fixes for 2.x-preview

### PR DESCRIPTION
**Pull request by bitsforthought**
See individual commits for details.

FYI @Yahnoosh @mhko @chaosrealm

### CODE REVIEW

**Comment by Yahnoosh** [comment]
LGTM :metal:

File: src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/ClassicTokenizer.cs [hunk 1]
```diff
@@ -40,8 +40,10 @@ public ClassicTokenizer(string name, int? maxTokenLength = default(int?))
         }
 
         /// <summary>
-        /// Gets or sets the maximum token length. Tokens longer than the
-        /// maximum length are split. Default is 255.
+        /// Gets or sets the maximum token length. Default is 255. Tokens
+        /// longer than the maximum length are split. The maximum token
+        /// length that can be used is 300 characters. Tokens longer than 300
+        /// characters are ignored.
```
Comment (Yahnoosh) [inline src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/ClassicTokenizer.cs]: This is not true for this tokenizer, just checked. We will not allow setting the limit to more than 300, as you wrote, and tokens longer than the limit will be split.

File: src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/StandardAnalyzer.cs [hunk 2]
```diff
@@ -40,8 +40,10 @@ public StandardAnalyzer(string name, int? maxTokenLength = default(int?), IList<
         }
 
         /// <summary>
-        /// Gets or sets the maximum token length. Tokens longer than the
-        /// maximum length are split. Default is 255.
+        /// Gets or sets the maximum token length. Default is 255. Tokens
+        /// longer than the maximum length are split. The maximum token
+        /// length that can be used is 300 characters. Tokens longer than 300
```
Comment (Yahnoosh) [inline src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/StandardAnalyzer.cs]: This is not true for this tokenizer, just checked. We will not allow setting the limit to more than 300, as you wrote, and tokens longer than the limit will be split.

File: src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/StandardTokenizer.cs [hunk 3]
```diff
@@ -39,8 +39,10 @@ public StandardTokenizer(string name, int? maxTokenLength = default(int?))
         }
 
         /// <summary>
-        /// Gets or sets the maximum token length. Tokens longer than the
-        /// maximum length are split. Default is 255.
+        /// Gets or sets the maximum token length. Default is 255. Tokens
+        /// longer than the maximum length are split. The maximum token
+        /// length that can be used is 300 characters. Tokens longer than 300
+        /// characters are ignored.
```
Comment (Yahnoosh) [inline src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/StandardTokenizer.cs]: This is not true for this tokenizer, just checked. We will not allow setting the limit to more than 300, as you wrote, and tokens longer than the limit will be split.

File: src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/UaxUrlEmailTokenizer.cs [hunk 4]
```diff
@@ -39,8 +39,10 @@ public UaxUrlEmailTokenizer(string name, int? maxTokenLength = default(int?))
         }
 
         /// <summary>
-        /// Gets or sets the maximum token length. Tokens longer than the
-        /// maximum length are split. Default is 255.
+        /// Gets or sets the maximum token length. Default is 255. Tokens
+        /// longer than the maximum length are split. The maximum token
+        /// length that can be used is 300 characters. Tokens longer than 300
+        /// characters are ignored.
```
Comment (Yahnoosh) [inline src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/UaxUrlEmailTokenizer.cs]: This is not true for this tokenizer, just checked. We will not allow setting the limit to more than 300, as you wrote, and tokens longer than the limit will be split.

File: src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/ClassicTokenizer.cs [hunk 5]
```diff
@@ -40,8 +40,10 @@ public ClassicTokenizer(string name, int? maxTokenLength = default(int?))
         }
 
         /// <summary>
-        /// Gets or sets the maximum token length. Tokens longer than the
-        /// maximum length are split. Default is 255.
+        /// Gets or sets the maximum token length. Default is 255. Tokens
+        /// longer than the maximum length are split. The maximum token
+        /// length that can be used is 300 characters. Tokens longer than 300
```
Comment (Yahnoosh) [inline src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/ClassicTokenizer.cs]: It's not true that tokens longer than 300 will be ignored, I've just checked. Please change the message to : 

Gets or sets the maximum token length. Default is 255. Tokens longer than the maximum length are split. The maximum token length that can be used is 300 characters.

Comment (brjohnstmsft) [inline src/Search/Microsoft.Azure.Search/GeneratedSearchService/Models/ClassicTokenizer.cs]: Ok, I'll update and send another iteration.

**Comment by Yahnoosh** [comment]
LGTM :metal:

**Pull request merged**